kvbc/src/merkle_tree_db_adapter.cpp

```diff
@@ -137,16 +138,25 @@ DBAdapter::DBAdapter(const std::shared_ptr<IDBClient> &db,
   }
 }
 
+std::optional<std::pair<Value, BlockId>> DBAdapter::getValueForNonProvableKey(const Key &key,
+                                                                              const BlockId &blockVersion) const {
+  const auto &blockKey = DBKeyManipulator::genNonProvableBlockDbKey(blockVersion, key);
```

Comment:
Nit: I think it will be more readable if we don't bind a reference to a temporary, i.e. just do `const auto blockKey`. Looking at the const ref, it looks like we are returning some constant internal object that way.

Reply:
The binding to a const reference guarantees no-copy according to the standard, i.e. better performance.
Since the object is only passed by reference and never copied, I think we can use it.

Reply:
In practice, there will be RVO and there won't be any copies. I am willing to trade for readability :)